src/pandroid/app/src/main/java/com/panda3ds/pandroid/utils/PathUtils.java

```diff
@@ -0,0 +1,101 @@
+package com.panda3ds.pandroid.utils;
+
+import android.content.ContentUris;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Environment;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+
+public class PathUtils {
+	public static String getPath(final Context context, final Uri uri) {
+		// DocumentProvider
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT && DocumentsContract.isDocumentUri(context, uri)) {
+			if (isExternalStorageDocument(uri)) { // ExternalStorageProvider
+				final String docId = DocumentsContract.getDocumentId(uri);
+				final String[] split = docId.split(":");
+				final String type = split[0];
+				String storageDefinition;
+
+				if ("primary".equalsIgnoreCase(type)) {
+					return Environment.getExternalStorageDirectory() + "/" + split[1];
+
```

Comment:
This empty line should be removed, unorthodox spacing (since blank lines should prevent walls of text but in this case there's literally nothing else in the block)

Reply:
Done - note this file will be removed when SAF is done